### PR TITLE
Update and rename Shape.rb to shape.rb

### PR DESCRIPTION
**Pull request by AhsanAkbar252**
Created separated file for shape. Changed shape to abstract class.

### CODE REVIEW

**Comment by AhsanAkbar252** [comment]
Separated files for shapes and made other changes

File: circle.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  attr_accessor :radius
+
+  PI = 3.14
+  def initialize(r)
+    @radius = r
+  end
+
+  def perimeter
+    # C=2πr
+    2 * PI * @radius
```
Comment (suleman-uzair) [inline circle.rb:12]: use direct 'radius' here because it is an attr_accessor

File: circle.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require_relative 'shape'
+class Circle < Shape
+  attr_accessor :radius
+
+  PI = 3.14
```
Comment (suleman-uzair) [inline circle.rb:5]: use freeze

File: circle.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  def initialize(r)
```
Comment (suleman-uzair) [inline circle.rb:6]: use descriptive variable names -> (r)

File: shape.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+class Shape
+  def area
+    raise NoMethodError, "#{self.class} prea method needs to be implemented."
```
Comment (suleman-uzair) [inline shape.rb:3]: area*

File: triangle.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    @side2 = side2
+    @side3 = side3
+  end
+
+  def perimeter
+    @side1 + @side2 + @side3
+  end
+
+  def area
+    @semi_perimeter = getPerimeter / 2  # s=(a + b + c)/2.
```
Comment (suleman-uzair) [inline triangle.rb:16]: @semi_perimeter, we dont need to use instance variable, 
@semi_perimeter -> perimeter

**Comment by suleman-uzair** [comment]
PR description is empty

File: circle.rb (hunk 2 above)
Comment (suleman-uzair) [reply circle.rb:5]: also read about it

**Comment by suleman-uzair** [comment]
Cycle 1 - Done

**Pull request merged**